File: ios/Packages/PhotocellCore/Sources/PhotocellCore/Nanos.swift
import Foundation

/// Tempo em nanossegundos inteiros do relógio do sensor (PTS do `CMSampleBuffer` convertido para
/// escala 1e9 no iOS; `SENSOR_TIMESTAMP` no Android). Nunca use relógio de CPU/thread aqui.
public typealias Nanos = Int64

public let nsPerSecond: Int64 = 1_000_000_000

/// E|X−Y| = 2σ/√π para X,Y ~ N(·, σ): converte o ΔY médio da calibração em σ por pixel.
public let meanAbsDiffToSigma: Double = 1.1283791670955126

@inline(__always)
func floorDiv(_ a: Int64, _ b: Int64) -> Int64 {
    let q = a / b
    return (a % b != 0 && ((a < 0) != (b < 0))) ? q - 1 : q
}

File: android/core/src/main/kotlin/br/com/tportooliveira/fotocelula/core/Nanos.kt
package br.com.tportooliveira.fotocelula.core

/**
 * Tempo em nanossegundos inteiros do relógio do sensor (SENSOR_TIMESTAMP no Android,
 * PTS convertido para escala 1e9 no iOS). Nunca use relógio de CPU/thread aqui.
 */
typealias Nanos = Long

const val NS_PER_SEC: Long = 1_000_000_000L

/** E|X-Y| = 2σ/√π para X,Y ~ N(·, σ): converte ΔY médio (calibração) em σ por pixel. */
const val MEAN_ABS_DIFF_TO_SIGMA: Double = 1.1283791670955126
